### PR TITLE
Fix externalLinkPWATest UI test on Firebase

### PR DESCRIPTION
**Pull request by AndiAJ**
Fix `externalLinkPWATest` ✅ successfully ran **100x** on Firebase.

There are 2 reasons to use another search provider:
• Google is sometimes blocking our requests, read an article that duckduckgo could be the solution to overcome the "unusual traffic" problem.
image

• When we run the UI tests locally for Beta or RC builds, we always need to manually dismiss Google's cookie consent 

<details>

<summary> 📈 Results (dropdown)</summary>

matrix | result | logs | details
-- | -- | -- | --
matrix-1x9kjquapvv44 | success | logs | 1 test cases passed
matrix-1s7xb4jqqu9jy | success | logs | 1 test cases passed
matrix-zr8fpsouu72na | success | logs | 1 test cases passed
matrix-jrka5kes26q3a | success | logs | 1 test cases passed
matrix-ec67b5vyx7kla | success | logs | 1 test cases passed
matrix-3fo6gptyax2g9 | success | logs | 1 test cases passed
matrix-197xwo9u9x5po | success | logs | 1 test cases passed
matrix-bj85y2dj2qqsa | success | logs | 1 test cases passed
matrix-183hy85kkw1bg | success | logs | 1 test cases passed
matrix-2aam9s9jjjphm | success | logs | 1 test cases passed
matrix-2i07oz74ajngg | success | logs | 1 test cases passed
matrix-2ingztwqxjruq | success | logs | 1 test cases passed
matrix-1hxlmjjgfaikc | success | logs | 1 test cases passed
matrix-3qdv1k5bsywkc | success | logs | 1 test cases passed
matrix-3tb0mp3zjfr6a | success | logs | 1 test cases passed
matrix-1j2ubilzcopiz | success | logs | 1 test cases passed
matrix-kr3hhfju08l8a | success | logs | 1 test cases passed
matrix-1ctqf3ywazy1e | success | logs | 1 test cases passed
matrix-10mm0i3c9cx2m | success | logs | 1 test cases passed
matrix-3ncuheyijgk7i | success | logs | 1 test cases passed
matrix-1332ghce3t57p | success | logs | 1 test cases passed
matrix-6ecgo9aj3ukxa | success | logs | 1 test cases passed
matrix-2cspryaqeg7ox | success | logs | 1 test cases passed
matrix-15e2wi2agjx8k | success | logs | 1 test cases passed
matrix-1r4qdf67qhe5t | success | logs | 1 test cases passed
matrix-3rx06ri9yj03z | success | logs | 1 test cases passed
matrix-wu1igpwrndn1a | success | logs | 1 test cases passed
matrix-277d9ory4k2ht | success | logs | 1 test cases passed
matrix-2rqxca7s5izsb | success | logs | 1 test cases passed
matrix-v37eq6ums799a | success | logs | 1 test cases passed
matrix-muc09m53p86ra | success | logs | 1 test cases passed
matrix-ubrz61u5h1l4a | success | logs | 1 test cases passed
matrix-3gapwd0vsbkt2 | success | logs | 1 test cases passed
matrix-1znfufq77lo5h | success | logs | 1 test cases passed
matrix-2zmugwz4oftr3 | success | logs | 1 test cases passed
matrix-nqhz8xbszy3ra | success | logs | 1 test cases passed
matrix-ohgvfqvv2a4aa | success | logs | 1 test cases passed
matrix-1g0zzh2niwiea | success | logs | 1 test cases passed
matrix-258amq9tot98x | success | logs | 1 test cases passed
matrix-2g9f1vsfoy87a | success | logs | 1 test cases passed
matrix-1fmkmk1t9on3o | success | logs | 1 test cases passed
matrix-3d27kc7ig9sui | success | logs | 1 test cases passed
matrix-2kyxugxlrhyar | success | logs | 1 test cases passed
matrix-382w1ehma54ew | success | logs | 1 test cases passed
matrix-2prylam56nhht | success | logs | 1 test cases passed
matrix-1refsssxw2tua | success | logs | 1 test cases passed
matrix-2p5a0g67t6e7e | success | logs | 1 test cases passed
matrix-2ca71ch0qofom | success | logs | 1 test cases passed
matrix-1egapy7pwn7rz | success | logs | 1 test cases passed
matrix-3qpmgbb2fnmse | success | logs | 1 test cases passed
matrix-3lq5yx2ip34xs | success | logs | 1 test cases passed
matrix-1vbobwv2rqqrz | success | logs | 1 test cases passed
matrix-d2kw5qmf0aspa | success | logs | 1 test cases passed
matrix-3kdwutdci8cne | success | logs | 1 test cases passed
matrix-1z6xibd9nguwb | success | logs | 1 test cases passed
matrix-2f0s2b2c8ah8o | success | logs | 1 test cases passed
matrix-3groemj63jowz | success | logs | 1 test cases passed
matrix-2sx7f0m95gg5n | success | logs | 1 test cases passed
matrix-2j8o35rpmhjvt | success | logs | 1 test cases passed
matrix-350oy1fhqdzak | success | logs | 1 test cases passed
matrix-34ggl4l6ccv5a | success | logs | 1 test cases passed
matrix-300dj5m404h43 | success | logs | 1 test cases passed
matrix-1zlow0xl2kckw | success | logs | 1 test cases passed
matrix-1txab8jof45zv | success | logs | 1 test cases passed
matrix-159xv3wmweywv | success | logs | 1 test cases passed
matrix-37z9n322dy941 | success | logs | 1 test cases passed
matrix-249n1q6pq7fp1 | success | logs | 1 test cases passed
matrix-3j2jv64ykb87l | success | logs | 1 test cases passed
matrix-h8olpn277xr9a | success | logs | 1 test cases passed
matrix-wqc4iq4c8m06a | success | logs | 1 test cases passed
matrix-3jbwr2zhzyw93 | success | logs | 1 test cases passed
matrix-s6d8cliioqlwa | success | logs | 1 test cases passed
matrix-3fg46wq8516zd | success | logs | 1 test cases passed
matrix-2zuwa4u45qpld | success | logs | 1 test cases passed
matrix-10rj2ofnim6yh | success | logs | 1 test cases passed
matrix-3ls6fxk7sgejk | success | logs | 1 test cases passed
matrix-162he8v2rd8sq | success | logs | 1 test cases passed
matrix-2utysmg0dmdjk | success | logs | 1 test cases passed
matrix-2paslazul06iy | success | logs | 1 test cases passed
matrix-3o7igz6vlzq6z | success | logs | 1 test cases passed
matrix-329crm4cpygas | success | logs | 1 test cases passed
matrix-qj80ozs6ku66a | success | logs | 1 test cases passed
matrix-13dfpa5mh6966 | success | logs | 1 test cases passed
matrix-uw7gx7oo7ywqa | success | logs | 1 test cases passed
matrix-3qnjk2bx1m7q1 | success | logs | 1 test cases passed
matrix-9roki4ftywyra | success | logs | 1 test cases passed
matrix-28r1sw9itp55d | success | logs | 1 test cases passed
matrix-pz6g3ju0ck6ca | success | logs | 1 test cases passed
matrix-bkil97mhgr3pa | success | logs | 1 test cases passed
matrix-1u5kqt3feowrf | success | logs | 1 test cases passed
matrix-3antel8c3omlr | success | logs | 1 test cases passed
matrix-2bfa4r9jkcqks | success | logs | 1 test cases passed
matrix-2wu731r8hxcwf | success | logs | 1 test cases passed
matrix-2aj8135jruff4 | success | logs | 1 test cases passed
matrix-16n01l41ocyuz | success | logs | 1 test cases passed
matrix-3rpjx6t9achwc | success | logs | 1 test cases passed
matrix-18ee204t1p1xw | success | logs | 1 test cases passed
matrix-3hys9ncszp6kl | success | logs | 1 test cases passed
matrix-2a2fcpzabzob1 | success | logs | 1 test cases passed
matrix-1kxm5z7u8m9r0 | success | logs | 1 test cases passed
matrix-1e9jwb1abe6hm | success | logs | 1 test cases passed

</details>

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt [hunk 1]
```diff
@@ -27,7 +27,7 @@ import org.mozilla.fenix.ui.robots.searchScreen
 class CustomTabsTest {
     private lateinit var mockWebServer: MockWebServer
     private val customMenuItem = "TestMenuItem"
-    private val externalLinksPWAPage = "https://mozilla-mobile.github.io/testapp/externalLinks.html"
+    private val externalLinksPWAPage = "https://andiaj.github.io/testapp/v2.0/externalLinks.html"
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt:30]: I think you forgot to change back to the mozilla-mobile domain

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 2]
```diff
@@ -570,13 +570,15 @@ class BrowserRobot {
         tabCrashRestoreButton.click()
     }
 
-    fun fillAndSubmitGoogleSearchQuery(searchString: String) {
+    fun fillAndSubmitDuckDuckGoSearchQuery(searchString: String) {
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:573]: could you provide some insight into why this change is needed, please? Maybe also on the github issue or PR description?

File: app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt:30]: Sure, first I want to get approval for the versioning update.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 2 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:573]: Thanks for clarifying. That's the reason I asked, because I know that duckDuckGo also blocks suspicious traffic from my experience with adding custom search engines. I propose to use one of the testapp pages as external link (even if it doesn't cover the manual test steps 100%).

**Comment by AndiAJ** [comment]
As per @sv-ohorvath 's suggestion I've changed the hypertext reference for "External link" to https://mozilla-mobile.github.io/testapp/downloads re-ran the affected tests and successfully passed 50x ✅

<details>

<summary> 📈 Results (dropdown)</summary>

matrix | result | logs | details
-- | -- | -- | --
matrix-3t6sdtnh9dxy5 | success | logs | 2 test cases passed
matrix-3kl4q51dpl9ro | success | logs | 2 test cases passed
matrix-14miyoacshkjb | success | logs | 2 test cases passed
matrix-9266ubiuk27ca | success | logs | 2 test cases passed
matrix-fnynf5bjunb1a | success | logs | 2 test cases passed
matrix-263ovgpn9q5bi | success | logs | 2 test cases passed
matrix-2s7q87ph9k1qo | success | logs | 2 test cases passed
matrix-17tjrpzudaj9e | success | logs | 2 test cases passed
matrix-5i28oocpicd6a | success | logs | 2 test cases passed
matrix-1ehmy90y2u4ux | success | logs | 2 test cases passed
matrix-fvnsqc3v7nf0a | success | logs | 2 test cases passed
matrix-2dmrgca0tioff | success | logs | 2 test cases passed
matrix-1enu7pd1da6in | success | logs | 2 test cases passed
matrix-18i43pn3fo1oe | success | logs | 2 test cases passed
matrix-17vthqcqx4di1 | success | logs | 2 test cases passed
matrix-6ez1xqe8qmc1a | success | logs | 2 test cases passed
matrix-reuu8rvjyds9a | success | logs | 2 test cases passed
matrix-15nju7d5bfdt6 | success | logs | 2 test cases passed
matrix-3eu0xhf481j20 | success | logs | 2 test cases passed
matrix-16k893sdaor3m | success | logs | 2 test cases passed
matrix-holmidut4mosa | success | logs | 2 test cases passed
matrix-23j4pvfeovuow | success | logs | 2 test cases passed
matrix-zwql5f034an7a | success | logs | 2 test cases passed
matrix-3vygosvcotiez | success | logs | 2 test cases passed
matrix-birk7mqdyydta | success | logs | 2 test cases passed
matrix-o4tepomc7qw3a | success | logs | 2 test cases passed
matrix-29pzh8zfyxqbi | success | logs | 2 test cases passed
matrix-3vh73wwtsij7a | success | logs | 2 test cases passed
matrix-363tta0s7gqf5 | success | logs | 2 test cases passed
matrix-33g0oltjj5tdc | success | logs | 2 test cases passed
matrix-1n0hkelsiie3r | success | logs | 2 test cases passed
matrix-gxfv20u6zmvva | success | logs | 2 test cases passed
matrix-2rmrypt8fozv7 | success | logs | 2 test cases passed
matrix-2wzbx34e7ak89 | success | logs | 2 test cases passed
matrix-1k5q70htwx6w5 | success | logs | 2 test cases passed
matrix-xr26iix9p6u1a | success | logs | 2 test cases passed
matrix-2ciz8yfk54w7s | success | logs | 2 test cases passed
matrix-21mx9xywoe58n | success | logs | 2 test cases passed
matrix-1ef88mv9kbxvj | success | logs | 2 test cases passed
matrix-31nhl8q0dm27y | success | logs | 2 test cases passed
matrix-jgnjd60xhmqva | success | logs | 2 test cases passed
matrix-3ut9kn4w9fuhi | success | logs | 2 test cases passed
matrix-255jy3lfhlfk4 | success | logs | 2 test cases passed
matrix-17a0b9ja2nj0w | success | logs | 2 test cases passed
matrix-1nr72m8jvx2br | success | logs | 2 test cases passed
matrix-2sszt4dn3l34c | success | logs | 2 test cases passed
matrix-1p66lh3mim51s | success | logs | 2 test cases passed
matrix-2kqcq8ke8ici5 | success | logs | 2 test cases passed
matrix-1y00wx4zcr44i | success | logs | 2 test cases passed
matrix-lwob3xu74youa | success | logs | 2 test cases passed

</details>

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt]: Done!

File: app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt [hunk 3]
```diff
@@ -27,7 +27,10 @@ import org.mozilla.fenix.ui.robots.searchScreen
 class CustomTabsTest {
     private lateinit var mockWebServer: MockWebServer
     private val customMenuItem = "TestMenuItem"
-    private val externalLinksPWAPage = "https://mozilla-mobile.github.io/testapp/externalLinks.html"
+    /* Updated externalLinks.html to v2.0,
+       changed the hypertext reference to mozilla-mobile.github.io/testapp/downloads for "External link"
+     */
+    private val externalLinksPWAPage = "https://andiaj.github.io/testapp/v2.0/externalLinks.html"
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt:33]: Needs an updated URL

File: app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt [hunk 4]
```diff
@@ -54,6 +57,7 @@ class CustomTabsTest {
     @SmokeTest
     @Test
     fun customTabsOpenExternalLinkTest() {
+        val externalLinkURL = "mozilla-mobile.github.io/testapp/downloads"
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt:60]: `https://`

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt [hunk 5]
```diff
@@ -37,7 +40,7 @@ class PwaTest {
     @SmokeTest
     @Test
     fun externalLinkPWATest() {
-        val customTabTitle = "Google"
+        val externalLinkURL = "mozilla-mobile.github.io/testapp/downloads"
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt:43]: `https://`

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt (hunk 3 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt]: Done!

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt [hunk 6]
```diff
@@ -16,7 +16,10 @@ import org.mozilla.fenix.ui.robots.navigationToolbar
 
 class PwaTest {
     private val featureSettingsHelper = FeatureSettingsHelper()
-    private val externalLinksPWAPage = "https://mozilla-mobile.github.io/testapp/externalLinks.html"
+    /* Updated externalLinks.html to v2.0,
+       changed the hypertext reference to mozilla-mobile.github.io/testapp/downloads for "External link"
+     */
+    private val externalLinksPWAPage = "https://andiaj.github.io/testapp/v2.0/externalLinks.html"
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt]: Done !

File: app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt (hunk 4 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/CustomTabsTest.kt]: Done!

File: app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt (hunk 5 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/PwaTest.kt]: Done!
